packages/xrpc-server/src/stream/websocket-keepalive.ts

```diff
@@ -0,0 +1,146 @@
+import { SECOND, wait } from '@atproto/common'
+import { WebSocket, ClientOptions } from 'ws'
+import { streamByteChunks } from './stream'
+import { CloseCode, DisconnectError } from './types'
+
+export class WebSocketKeepAlive {
+  public ws: WebSocket | null = null
+  public initialSetup = true
+  public reconnects: number | null = null
+  public isAlive = false
+  public heartbeatInterval: NodeJS.Timer | null
+
+  constructor(
+    public opts: ClientOptions & {
+      getUrl: () => Promise<string>
+      maxReconnectSeconds?: number
+      signal?: AbortSignal
+      heartbeatInterval?: number
```

Comment:
Kind of funny naming conundrum here, where "interval" is sort of used in different senses in the option (a duration) versus the class property (a node timer).  Maybe just making the option `heartbeatIntervalMs` or something like that would be a way to distinguish them.

Reply:
nice i like it 👍